about/release_policy.rst

```diff
@@ -158,3 +195,55 @@ The 3.6 release is still planned and should be the last stable branch of Godot
 as long as users still need it (due to missing features in Godot 4.x, or
 having published games which they need to keep updating for platform
 requirements).
+
+What are the criteria for compatibility across engine versions?
+---------------------------------------------------------------
+
+.. note::
+
+    This section is intended to be used by contributors to determine which
+    changes are safe for a given release. The list is not exhaustive; it only
+    outlines the most common situations encountered during Godot's development.
+
+The following changes are acceptable in patch releases:
```

Comment:
Worth mentioning in some form that we tend to be more conservative with the fixes we allow in each subsequent patch release. So e.g. 4.0.1 may receive more impactful fixes than 4.0.4 would.